src/wallet/crypter.cpp

```diff
@@ -290,7 +290,7 @@ bool CCryptoKeyStore::Unlock(const CKeyingMaterial& vMasterKeyIn)
 bool CCryptoKeyStore::SetHDSeed(const HDSeed& seed)
 {
     {
-        LOCK(cs_SpendingKeyStore);
+        LOCK2(cs_KeyStore, cs_SpendingKeyStore);
```

Comment:
This change should not be necessary yet, because only Sapling spending keys are derived from the HD seed. But if we ever migrated transparent addresses to HD, then this would probably become necessary.